src/domain.rs

```diff
@@ -66,7 +66,8 @@ pub enum TrustLevel {
 #[derive(Debug, Clone)]
 pub struct Executable<D: Domain> {
     pub expr: Expr,
-    pub evals: RefCell<HashMap<(Id,Vec<LazyVal<D>>), Val<D>>>, // from (node,env) to result
+    #[allow(clippy::type_complexity)]
+    pub evals: RefCell<HashMap<(Id, Vec<LazyVal<D>>), Val<D>>>, // from (node,env) to result
```

Comment:
clippy doesn't like this type. I don't blame it. I've disabled the warning for now, but we should probably consider giving this type a more legible/explicit definition.